lib/tasks/load_test.rake

```diff
@@ -0,0 +1,205 @@
+# * Use a specific user to run these tests
+# * Move parameters to a config file for more flexibility and clarity.
+
+namespace "load_test" do
+  def parse_samples_from_s3_path(s3, s3_bucket, s3_prefix)
+    # Some parts were adapted from samples_helper::parsed_samples_for_s3_path
+    # It is hard to import our helpers here given all the dependencies - also not a best practice
+    resp = s3.list_objects_v2(
+      bucket: s3_bucket,
+      prefix: s3_prefix,
+      max_keys: 100
+    )
+    samples = {}
+    resp.contents.each do |object|
+      filename = File.basename(object.key)
+      if (match = InputFile::BULK_FILE_PAIRED_REGEX.match(filename))
+        name = match[1]
+        read_idx = match[2].to_i - 1
+      elsif (match = InputFile::BULK_FILE_SINGLE_REGEX.match(filename))
+        name = match[1]
+        read_idx = 0
+      else
+        next
+      end
+
+      samples[name] ||= {
+        name: name,
+        input_files_attributes: [],
+      }
+      samples[name][:input_files_attributes][read_idx] = {
+        name: filename,
+        source: "s3://#{s3_bucket}/#{object.key}",
+        source_type: InputFile::SOURCE_TYPE_S3,
+      }
+    end
+    return samples
+  end
+
+  desc "start a load test with <total_samples> samples select round-robin fashion from <samples_path>"
+  task :start, [:total_samples, :s3_bucket, :samples_path, :project_name, :user_id] => :environment do |_, args|
+    Rails.logger = Logger.new(STDOUT)
+
+    Rails.logger.info("Starting a load test in #{Rails.env}")
+    Rails.logger.info("Args: #{args}")
+
+    begin
+      total_samples = Integer(args.total_samples)
+    rescue
+      Rails.logger.error("Total samples is not an integer: #{args.total_samples}.")
+      abort("Load test failed")
+    end
+
+    user = User.find_by(id: args.user_id)
+    unless user
+      Rails.logger.error("Please specify an admin user.")
+      abort("Load test failed")
+    end
+    unless user.admin?
+      Rails.logger.error("Please specify an admin user. User was: name='#{user.name}' admin=#{user.admin?}")
+      abort("Load test failed")
+    end
+
+    project = Project.find_by(name: args.project_name)
+    unless !project || Sample.where(project: project).empty?
+      Rails.logger.error("Please specify a unexistent or empty project. Project '#{project.name}' has samples.")
+      abort("Load test failed")
+    end
+
+    # Load metadata.csv
+    s3 = Aws::S3::Client.new
+    metadata_file_path = File.join(args.samples_path, "metadata.csv")
+    begin
+      resp_metadata = s3.get_object(
+        bucket: args.s3_bucket,
+        key: metadata_file_path
+      )
+      metadata_samples = CSV.parse(resp_metadata.body.read, headers: true).map(&:to_h)
+    rescue Aws::S3::Errors::NoSuchKey
+      Rails.logger.error("Metadata file not found [bucket=#{args.s3_bucket} path=#{metadata_file_path}]")
+      abort("Load test failed")
+    end
```

Comment:
How do I generate this metadata file? Can you include a link to a sample file or format description?

Reply:
The metadata file template can be downloaded from IDseq. The current s3 folder used for testing this can be used for staging and prod, so you won't have to generate this.
Added a link in the instructions above.